### DISCUSSION TITLE
fromJSON

### DISCUSSION THREAD

**Discussion by scarf005** [rimbu-org/rimbu#208] [Ideas]
while there's [`toJSON`](https://rimbu.org/api/rimbu/collection-types/set-custom/VariantSetBase/interface#tojson) to represent rimbu data in JSON format, there isn't matching `fromJSON` to reverse it.

having deserialization built-in will help sending and caching rimbu data.

**Comment by vitoke** [comment]
I completely agree. Some time ago I started [this branch](https://github.com/rimbu-org/rimbu/tree/feature/add-serializer) which is probably beyond repair now. The main issue for serialization is how to deal with the concept of `Context`. If I create a custom context, for example, for a `HashMap` with my own hashing and equality functions, how can we know this when deserializing data. Or can we just ignore that problem? I believe I started adding ids to context so that at least locally you could know which context the data belongs to, but since contexts can sometimes also include other contexts it quickly becomes quite complex. What are your thoughts?
